### PR TITLE
Profiles: Various fixes

### PR DESCRIPTION
**Pull request by jxom**
This PR fixes an issue with inconsistent cover photo uploads (cover photo appearing again when it was removed).

Also fixes a few instances where we were experiencing infinite loops in some places

### CODE REVIEW

File: src/components/ens-registration/RegistrationCover/RegistrationCover.tsx [hunk 1]
```diff
@@ -38,10 +39,18 @@ export default function RegistrationCover({
     values,
   } = useENSRegistrationForm();
 
+  const [coverUpdateAllowed, setCoverUpdateAllowed] = useState(true);
```
Comment (jxom) [inline src/components/ens-registration/RegistrationCover/RegistrationCover.tsx:42]: Taking logic from `RegistrationAvatar` here.

File: src/hooks/useENSRegistration.ts [hunk 2]
```diff
@@ -146,7 +146,8 @@ export default function useENSRegistration({
 
   useEffect(() => {
     dispatch(ensRedux.setChangedRecords(accountAddress, changedRecords));
-  }, [accountAddress, changedRecords, dispatch]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [accountAddress, JSON.stringify(changedRecords), dispatch]);
```
Comment (jxom) [inline src/hooks/useENSRegistration.ts:150]: This was causing an infinite loop

File: src/hooks/useENSRegistrationCosts.ts [hunk 3]
```diff
@@ -84,10 +84,11 @@ export default function useENSRegistrationCosts({
     return registrationParameters?.name !== name && name.length > 2;
   }, [name, registrationParameters?.name]);
 
-  const [debouncedChangedRecords] = useDebounce(
-    registrationParameters?.changedRecords || {},
-    500
+  const changedRecords = useMemo(
```
Comment (jxom) [inline src/hooks/useENSRegistrationCosts.ts:87]: This was causing an infinite loop

File: src/screens/ENSConfirmRegisterSheet.js [hunk 4]
```diff
@@ -306,9 +306,11 @@ export default function ENSConfirmRegisterSheet() {
       setSendReverseRecord(!registrationCostsData?.hasReverseRecord);
   }, [registrationCostsData?.hasReverseRecord]);
 
-  useFocusEffect(() => {
-    blurFields();
-  });
+  useFocusEffect(
```
Comment (jxom) [inline src/screens/ENSConfirmRegisterSheet.js:309]: This was causing an infinite loop

**Review by benisgold — APPROVED** [review]
all my homies hate infinite loops

**Review by estebanmino — APPROVED** [review]
full 🐐  mode

**Pull request merged**